web/tls_config.go

```diff
@@ -51,6 +52,8 @@ type TLSConfig struct {
 	MinVersion               TLSVersion `yaml:"min_version"`
 	MaxVersion               TLSVersion `yaml:"max_version"`
 	PreferServerCipherSuites bool       `yaml:"prefer_server_cipher_suites"`
+	// regular expression to match the SAN DNS entries of the client cert
+	ClientCertAllowedSanDNSRegex string `yaml:"client_cert_allowed_san_dns"`
```

Comment:
Minor nit, SANs can be a variety of things, not just DNS names. Let's just simplify this.

How about this:

```suggestion
	ClientAllowedSanRegex string `yaml:"client_allowed_san_regex"`
```